x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/MutableSearchResponse.java

```diff
@@ -95,6 +106,9 @@ synchronized void updatePartialResponse(
         int reducePhase
     ) {
         failIfFrozen();
+        if (clusters.isCcsMinimizeRoundtrips()) {
+            return;
+        }
```

Comment:
This makes sense if we decide that we don't want to expose partial aggs results, much better than them disappearing once a remote cluster returned its response. I have a counter proposal: could we instead look into creating a SearchResponseMerger, on the fly, and add the partial SearchResponse together with the partial aggs and expose the result of the merge? That should work or am I missing something?

```
InternalAggregations reducedAggs = reducedAggsSource.get();
reducedAggsSource = () -> reducedAggs;
SearchResponse searchResponse = buildResponse(task.getStartTimeNanos(), reducedAggs);
SearchResponseMerger merger = new SearchResponseMerger();
merger.add(searchResponse);
merger.add(partialSearchResponse);
SearchResponse newSearchResponse = merger.getMergedResponse();
```

It feels like this should fit with the other comment above on delaying the merging to once results are requested.